### PR TITLE
istio: improve deep copy for ServiceAttribute

### PR DESCRIPTION
**Pull request by yingzhuivy**
**Please provide a description of this PR:**

our istiod CPU profile shows that 10% of time is spent on DeepCopy of ServiceAttribute, so removing the use of copystructure. We already have fuzz test and benchmark for Service DeepCopy, which covers this part since ServiceAttribute is part of Service.

Benchmark results:
(before)
```goos: darwin
goarch: amd64
pkg: istio.io/istio/pilot/pkg/model
cpu: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
BenchmarkServiceDeepCopy
BenchmarkServiceDeepCopy-16    	  132760	      8190 ns/op
PASS
```
(after)
```goos: darwin
goarch: amd64
pkg: istio.io/istio/pilot/pkg/model
cpu: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
BenchmarkServiceDeepCopy
BenchmarkServiceDeepCopy-16    	 1213035	      1019 ns/op
PASS
```

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -566,7 +566,49 @@ func (s *ServiceAttributes) DeepCopy() ServiceAttributes {
 	// Nested mutexes are configured to be ignored by copystructure.Copy.
 	// Disabling `go vet` warning since this is actually safe in this case.
 	// nolint: vet
-	return copyInternal(*s).(ServiceAttributes)
+	out := *s
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/service.go:569]: remove the above comment

**Review by howardjohn — APPROVED** [review]
https://github.com/istio/istio/pull/40987 tested it more, lgtm

File: pilot/pkg/model/service.go (hunk 1 above)
Comment (yingzhuivy) [reply pilot/pkg/model/service.go:569]: followed up in https://github.com/istio/istio/pull/40990

**Pull request merged**